bench/benches/operators.rs

```diff
@@ -0,0 +1,178 @@
+    unsafe { COUNTER += 1 };
+
+    unsafe {
+        batch.group_by(
+            |_p| COUNTER,
+            |groups| {
+                compose!(groups {
+                    0 => |group| {
+                        group.for_each(|_p| Ok(()))
+                    }
+                    _ => |group| {
+                        group.for_each(|_p| Ok(()))
+                    }
+                })
+            },
+        )
+    }
+}
+
+fn replace(batch: impl Batch<Item = Mbuf>) -> impl Batch<Item = Mbuf> {
+    batch.replace(|_p| Mbuf::new())
+}
+
+fn no_batch_replace(_mbuf: Mbuf) -> Mbuf {
+    Mbuf::new().unwrap()
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn filters_batch(c: &mut Criterion) {
+    let mut group = c.benchmark_group("operators::filter");
+
+    group.bench_function("operators::filter_true", |b| {
+        let s = v4_udp();
+        b.iter_proptest_polled(s, filter_true, BATCH_SIZE)
+    });
+
+    group.bench_function("operators::filter_false", |b| {
+        let s = v4_udp();
+        b.iter_proptest_polled(s, filter_false, BATCH_SIZE)
+    });
+
+    group.finish()
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn filter_map_vs_map_then_filter_batch(c: &mut Criterion) {
+    let mut group = c.benchmark_group("operators::filter_map_vs_map_then_filter");
+
+    group.bench_function("operators::filter_map", |b| {
+        let s = strategy::Union::new_weighted(vec![(9, v4_udp().boxed()), (1, v6_udp().boxed())]);
+        b.iter_proptest_polled(s, filter_map, BATCH_SIZE)
+    });
+
+    group.bench_function("operators::map_then_filter", |b| {
+        let s = strategy::Union::new_weighted(vec![(9, v4_udp().boxed()), (1, v6_udp().boxed())]);
+        b.iter_proptest_polled(s, map_then_filter, BATCH_SIZE)
+    });
+
+    group.finish()
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn map_batch(c: &mut Criterion) {
+    c.bench_function("operators::map", |b| {
+        let s = strategy::Union::new_weighted(vec![(9, v4_udp().boxed()), (1, v6_udp().boxed())]);
+        b.iter_proptest_polled(s, map, BATCH_SIZE)
+    });
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn group_by_batch(c: &mut Criterion) {
+    c.bench_function("operators::group_by", |b| {
+        let s = v4_udp();
+        b.iter_proptest_polled(s, group_by, BATCH_SIZE)
+    });
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn replace_batch(c: &mut Criterion) {
+    let mut group = c.benchmark_group("operators::replace_with_new_mbuf_vs_create_new_mbuf");
+
+    group.bench_function("operators::replace", |b| {
+        let s = v4_udp();
+        b.iter_proptest_polled(s, replace, BATCH_SIZE)
+    });
+
+    group.bench_function("operators::no_batch_replace", |b| {
+        let s = v4_udp();
+        b.iter_proptest_batched(s, no_batch_replace, BATCH_SIZE)
+    });
+
+    group.finish()
+}
+
+#[nb2::bench(mempool_capacity = 1023)]
+fn map_batches(c: &mut Criterion) {
+    let mut group = c.benchmark_group("operators::map_on_diff_batch_sizes");
```

Comment:
what is the expected outcome here with these?

Reply:
Just see how batch size has any affects on straight combinator.

Reply:
results seem interesting at least.

Reply:
ok I'm curious to see the results.

Reply:
<img width="1013" alt="Screen Shot 2019-12-03 at 1 12 56 PM" src="https://user-images.githubusercontent.com/202820/70077901-b9552500-15cf-11ea-95ca-8e370521a99d.png">

Reply:
I don't know what it's saying.